server/discussion/queries.js

```diff
@@ -120,8 +121,13 @@ export const createDiscussion = async (inputValues, user) => {
 		userId: user.id,
 		threadId: newThread.id,
 	});
+
+	const branchData = await Branch.findOne({
```

Comment:
Without a corresponding update to `permissions.js` this will allow anyone who knows the ID of the draft branch to write a members-only comment. I confess this is unlikely, but I would propose the following structure:

On the client, send `visibility: pubData.isRelease ? 'public' : 'members'` as part of the payload to the API request. Then pass `visibility` into `getPermissions()`, and check whether the logged-in user can create a discussion on the Pub with the given visibility.

This is closer to our new mental model of PubPub and the code we'll need to write to support user-controlled discussion visibility anyway. Branches are sort of a vestigial part of our IA now so we should try to avoid passing around references to them from here on out.

Reply:
Good catch. I made changes in line with your suggestion in e28b77d. Testing shows it setting the right `access` value and rejecting those who try to set `members` without authorization. 

I'm going to resolve and merge this, but feel free to continue the conversation if you recommend additional refactors to this.